File: tests/fixtures/python_simple/service.py
from typing import Optional
from models import Item, CreateItemRequest
from repository import ItemRepository


class DataService:
    def __init__(self):
        self._repository = ItemRepository()
        self._counter = 0

    def get_item(self, item_id: int) -> Optional[Item]:
        return self._repository.find_by_id(item_id)

    def create_item(self, request: CreateItemRequest) -> Item:
        self._counter += 1
        item = Item(
            id=self._counter,
            name=request.name,
            category=request.category,
            price=request.price,
            active=True,
        )
        self._repository.save(item)
        return item

    def delete_item(self, item_id: int) -> bool:
        existing = self._repository.find_by_id(item_id)
        if existing is None:
            return False
        self._repository.delete(item_id)
        return True

    def list_items(self, category: str = None, active_only: bool = True) -> list:
        items = self._repository.find_all()
        if category:
            items = [i for i in items if i.category == category]
        if active_only:
            items = [i for i in items if i.active]
        return items

    def update_item(self, item_id: int, updates: dict) -> Item:
        item = self._repository.find_by_id(item_id)
        if item is None:
            raise ValueError(f"Item {item_id} not found")
        for key, value in updates.items():
            if hasattr(item, key):
                setattr(item, key, value)
        self._repository.save(item)
        return item

    def search(self, query: str) -> list:
        items = self._repository.find_all()
        query_lower = query.lower()
        return [i for i in items if query_lower in i.name.lower()]

File: tests/fixtures/cpp_simple/service.cpp
#include "service.hpp"

DataService::DataService() : counter_(0) {}

std::string DataService::getItem(int id) const {
    auto it = store_.find(id);
    if (it != store_.end() && it->second.active) {
        return it->second.name;
    }
    return "";
}

int DataService::createItem(const std::string& name) {
    if (!isValidName(name)) {
        return -1;
    }
    counter_++;
    ItemRecord record;
    record.id = counter_;
    record.name = name;
    record.category = "default";
    record.active = true;
    store_[counter_] = record;
    return counter_;
}

bool DataService::deleteItem(int id) {
    auto it = store_.find(id);
    if (it == store_.end()) {
        return false;
    }
    it->second.active = false;
    return true;
}

std::vector<ItemRecord> DataService::listItems() const {
    std::vector<ItemRecord> result;
    for (const auto& pair : store_) {
        if (pair.second.active) {
            result.push_back(pair.second);
        }
    }
    return result;
}

bool DataService::updateItem(int id, const std::string& name) {
    auto it = store_.find(id);
    if (it == store_.end() || !it->second.active) {
        return false;
    }
    it->second.name = name;
    return true;
}

int DataService::count() const {
    int active = 0;
    for (const auto& pair : store_) {
        if (pair.second.active) active++;
    }
    return active;
}

bool DataService::isValidName(const std::string& name) const {
    return !name.empty() && name.length() <= 256;
}
